plugin/src/main/java/com/denizenscript/denizen/events/block/BlockSpreadsScriptEvent.java

```diff
@@ -58,7 +56,7 @@ public boolean couldMatch(ScriptPath path) {
 
     @Override
     public boolean matches(ScriptPath path) {
-        if (!runInCheck(path, location)) {
+        if (!runInCheck(path, event.getBlock().getLocation())) {

```

Comment:
the field being used in `matches` here makes it (somewhat) worthwhile to have. It's not a huge difference either way, but `matches` is called _at least once_ if not _several times_ for every firing of the event, so precalculating values can be worth the small memory/performance improvement. As opposed to `getContext` which in the average event, the average context tag is never used, thus precalculating them is usually a bad idea (except where the value is likely to change during event processing and thus needs an original value to be cached)

Reply:
Removed it since it's just a single location getter so didn't think it's worth storing in a field, but yeah fair enough - reverted that